src/sentry/integrations/utils/metrics.py

```diff
@@ -197,3 +203,48 @@ def __exit__(
                 if self.assume_success
                 else EventLifecycleOutcome.HALTED
             )
+
+
+class IntegrationPipelineViewType(Enum):
+    """A specific step in an integration's pipeline that is not a static page."""
+
+    # IdentityProviderPipeline
+    IDENTITY_PROVIDER = "IDENTITY_PROVIDER"
+
+    # GitHub
+    OAUTH_LOGIN = "OAUTH_LOGIN"
```

Comment:
Should these have integration name in them? Are they meant to be shared?

Reply:
we already pass in the integration name when initializing the metric class